### PR TITLE
Add op_mapper Register

### PR DESCRIPTION
**Pull request by thisjiang**
Now, we can add `OpMapper` by `CINN_REGISTER_OPMAPPER` macro like:

```
CINN_REGISTER_HELPER(sigmoid) { CINN_REGISTER_OPMAPPER(sigmoid, cinn::frontend::op_mappers::SigmoidOpMapper) }
```
And you must add `CINN_USE_REGISTER(sigmoid)` macor in  `use_op_mappers.h` header file, otherwise the new opmapper will link failed so that we cannot obtain register information. 

The `SigmoidOpMapper` is a function and it's Function prototype as following:
```
void SigmoidOpMapper(const paddle::cpp::OpDesc& op_desc, const OpMapperContext& ctx) {
   // check op right
  CHECK_EQ(op_desc.Input("X").size(), 1UL);
  CHECK_EQ(op_desc.Output("Out").size(), 1UL);

  // get input and output name
  auto x_name = op_desc.Input("X").front();
  auto out_name = op_desc.Output("Out").front();

  // get input variable by input name
  auto x   = ctx.GetVar(x_name);

  // get output variable returned by net builder
  auto out = ctx.builder_->sigmoid(x);

  // add to local variable map
  ctx.AddVar(out_name, out);

  // add to transform name map
  ctx.AddVarModelToProgramMap(out_name, out->id);
}
```

And we can get these registered op kernel by include `use_op_mappers.h` and using `OpMapperRegistry::Global()->Find(op_type)` function to get kernel. The `OpMapperRegistry` is inherited from `Registry` class and no new API added.

More Example see https://github.com/PaddlePaddle/CINN/pull/460 .

### CODE REVIEW

File: cinn/frontend/op_mapper_registry.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  hlir::framework::Scope* scope_{nullptr};
+  const common::Target& target_;
+  NetBuilder* builder_{nullptr};
+  absl::flat_hash_map<std::string, Variable>* var_map_{nullptr};
+  // map from var in Paddle model to var name in program.
+  absl::flat_hash_map<std::string, std::string>* var_model_to_program_map_{nullptr};
+};
+
+class OpMapperKernel {
```
Comment (Superjomn) [inline cinn/frontend/op_mapper_registry.h]: 这个是不是最好叫  OpMapper ?

OpMapperRegistry 用于 register  OpMapper

Kernel 这个概念放过来有点不太对应

Comment (thisjiang) [reply cinn/frontend/op_mapper_registry.h]: Done, 已修改

**Review by Superjomn — APPROVED** [review]
LGTM

File: cinn/frontend/op_mapper_registry.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        builder_(builder),
+        var_map_(var_map),
+        var_model_to_program_map_(var_model_to_program_map) {}
+
+  hlir::framework::Scope* scope_{nullptr};
+  const common::Target& target_;
+  NetBuilder* builder_{nullptr};
+  absl::flat_hash_map<std::string, Variable>* var_map_{nullptr};
+  // map from var in Paddle model to var name in program.
+  absl::flat_hash_map<std::string, std::string>* var_model_to_program_map_{nullptr};
```
Comment (wzzju) [inline cinn/frontend/op_mapper_registry.h:38]: 类的属性还是应该声明为private的，或者你这里类比较简单，可以声明为struct。
```suggestion
struct OpMapperContext {
  OpMapperContext() {
    // ...
  }
  NetBuilder* builder{nullptr};
};
```

File: cinn/frontend/op_mapper_registry.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  OpMapper() = default;
+
+  OpMapper& Set(const KernelFunc& kernel) {
+    kernel_ = kernel;
+    return *this;
+  }
+  void Run(const paddle::cpp::OpDesc& op_desc, const OpMapperContext& ctx) const { kernel_(op_desc, ctx); }
+
+  std::string name;
```
Comment (wzzju) [inline cinn/frontend/op_mapper_registry.h:53]: Why `name` is public?

Comment (thisjiang) [reply cinn/frontend/op_mapper_registry.h:53]: CINN注册基类 `Registry`会直接访问name成员变量，所以必须是public的

File: cinn/frontend/op_mapper_registry.h (hunk 2 above)
Comment (thisjiang) [reply cinn/frontend/op_mapper_registry.h:38]: 好的，我再提个PR修改下

**Pull request merged**